benchmarks/PNP/pnp_adaptive.cpp

```diff
@@ -320,7 +323,9 @@ int main()
 
     // set initial residual
     printf("\tupdate initial residual...\n"); fflush(stdout);
-    initial_residual = get_initial_residual(&L_pnp, &bc, &initialCation, &initialAnion, &initialPotential);
+    base_functions::LinearForm Lbase(V);
```

Comment:
for easier reading, wouldn't it be easier to put everything involving ``base_functions`` in the function call?  This way, it's not obvious why this is needed...